### ISSUE TITLE
the low sharpe of the eiie

### ISSUE THREAD

**Issue by rikocyan** [CaioSBC/RLPortfolio#10]
Hello！

I ran through the BR_stocks code, but the returns obtained through eiie were lower than those of the Best Stock strategy and Follow the Loser strategy.

![Image](https://github.com/user-attachments/assets/e1876371-4933-4e7e-8ec6-e25fcb58f692)

How can I modify the parameters so that I can get a better return and Sharpe ratio through eiie? Please, this is very important for my paper:(

**Comment by CaioSBC** [comment]
Hello @rikocyan!

This is a common issue of this algorithm, different experiments with the same hyperparameters can lead to completely different results. You can try to run more experiments until you achieve the best outcome (which is basically what I did in this example).

Recently, I have noticed that normalizing the data before creating the environment instead of during state generation is better (check [Issue #9](https://github.com/CaioSBC/RLPortfolio/issues/9#issuecomment-2756537909)), however, another problem appears: the agent usually gets stuck in a local minima and adopts the "follow the loser" strategy.

I'm still trying to figure out how to improve the reliability of this algorithm, but it's still an open issue in the field. If you have any ideas, feel free to try it and share with us.

**Comment by rikocyan** [comment]
Thanks for your reply!


Yesterday I set the parameters like this ↓
algo_eiie.train(episodes=60, val_period=5, val_gradient_steps=15, val_env=environment_test)



Then I got this good result&nbsp;↓



I think it may be that the val_gradient_steps is set too high, which leads to overfitting. Or maybe the val_period is set too high, which leads to the failure to detect the decline in model performance in time.


Anyway, thank you for your reply. This is my first time chatting with the author on GitHub. Wish you have a good day! :)




         原始邮件
         
       
发件人：Caio Costa ***@***.***&gt;
发件时间：2025年4月22日 11:07
收件人：CaioSBC/RLPortfolio ***@***.***&gt;
抄送：rikocyan ***@***.***&gt;, Mention ***@***.***&gt;
主题：Re: [CaioSBC/RLPortfolio] the low sharpe of the eiie (Issue #10)



CaioSBC left a comment (CaioSBC/RLPortfolio#10)

Hello @rikocyan!

This is a common issue of this algorithm, different experiments with the same hyperparameters can lead to completely different results. You can try to run more experiments until you achieve the best outcome (which is basically what I did in this example).

Recently, I have noticed that normalizing the data before creating the environment instead of during state generation is better (check Issue #9), however, another problem appears: the agent usually gets stuck in a local minima and adopts the "follow the loser" strategy.

I'm still trying to figure out how to improve the reliability of this algorithm, but it's still an open issue in the field. If you have any ideas, feel free to try it and share with us.
